common/utils.js

```diff
@@ -3,6 +3,61 @@
 
     angular.module('chaise.utils', ['chaise.errors'])
 
+    .constant("defaultChaiseConfig", {
```

Comment:
I realize I had suggested we should include the default value for things that don't necessarily have a default defined. But there are some cases where we don't want a default to bet set. 

This default object won't be populated in the case of search (you didn't make a change to call `setConfigJSON()` in search app (which is fine, we don't want to change search functionality). This means any property that is specific to search shouldn't be included in here.

I don't think we need to define the following properties:
- catalog (deprecated config option. Catalog is not actually used anywhere in the code. duplicate of defaultCatalog that replaces it. defaultCatalog should be undefined by default)
- feedbackUrl (I don't see it used in any of the apps, this might be deprecated at this point)
- helpurl (only used as a link in the tour code which is only in search. `''` is the same as `undefined` for that code)
- recordResource (this shouldn't even be configurable, deprecated)
- showBadgeCounts (only used in search app)
- tableThreshold (unused)
- showAllAttributes (only used in search app)
- dataBrowser (this is used to redirect for error messages. By default, if not set, we redirect to the root or `window.location.origin`. We don't want to force deployments to redirect to search instead of root... the default for this is wrong in configuration.md)
- hideSearchTextFacet (only used in search)
- maxColumns (only used in search)
- showUnfilteredResults (only used in search)
- sidebarPosition (only used in search)
- attributesSidebarHeading (the default is different from what is defined in the search code. I'm not sure what is the right solution, so we should not set this property. also it's only in search)
- searchPageSize (only used in search)
- customCSS (should not have a default value. if this is undefined by the user, it's assumed there is no custom CSS document)
- footerMarkdown (there is no default for this and if it's undefined, we don't show a footer. This change won't allow the footer to be NOT shown)
- maxRelatedTablesOpen (the place in the code that this is used, we want this to be "infinity" if undefined, not 15. just remove this property)
- plotViewEnabled (only used in search)
- recordUiGridEnabled, recordUiGridExportCSVEnabled, recordUiGridExportPDFEnabled (unused, not even defined in configuration.md)
- tour (only used in search. search e2e tests are no longer run)


Some of the default values are also set wrong:
- navbarBrand (should be `""` if not defined, not root)
- editRecord (default is false)
- deleteRecord (default is false)
- allowErrorDismissal (default is false)
- showFaceting (default is false)

@RFSH  can you verify these 2 lists?